### ISSUE TITLE
Documentation: Missing includes

### ISSUE THREAD

**Issue by Meriipu** [TorchCraft/TorchCraft#69]
I followed the setup guide for developers https://github.com/TorchCraft/TorchCraft/blob/master/docs/user/installation.md  and opened TorchCraft in Visual Studio 2013, but when I start the build it says BWAPI.h is missing.

I have no idea where to or how to include this, when I tried copying over the header files without doing much else I get a lot of unresolved external symbol errors so that is probably not how.

**Comment by ebetica** [comment]
I think the trick is to either install BWAPI if you haven't or set up the right PATH in your window environment variables. (set it to the directory where BWAPI.h is)

**Comment by Meriipu** [comment]
I added the include directories of bwapi czmq and libzmq to the INCLUDE path, though now it errors with 
```
------ Build started: Project: BWEnv, Configuration: Release Win32 ------
  main.cc
C:\Users\pix\Source\Repos\TorchCraft\BWEnv\VisualStudio\\..\src\main.cc(20): error C2280: 'Controller::Controller(const Controller &)': attempting to reference a deleted function
  c:\users\pix\source\repos\torchcraft\bwenv\include\controller.h(94): note: compiler has generated 'Controller::Controller' here
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped =========
'''

**Comment by ebetica** [comment]
https://github.com/TorchCraft/TorchCraft/blob/master/BWEnv/src/main.cc#L20

This is Controller::Controller(bool), are you sure you didn't change anything in the source?

Also, we're releasing a new version of TorchCraft soon, try the develop branch to see if it gives you any more luck.

**Comment by Meriipu** [comment]
I can try setting it up on the windows end from the beginning again. Somehow I have a ~Controller() in my source though I never changed that part.

**Comment by Meriipu** [comment]
No, I started from scratch but Visual Studio 2015 fails with the same error

**Comment by ebetica** [comment]
Are you using VS2013 or VS2015? As far as I know, torchcraft doesn't compile on 2015.

**Comment by Meriipu** [comment]
I have tried both

most recent attempt in VS2013:
1) Open VS2013 and clone BWAPI.git, TorchCraft.git, libzmq.git, and czmq.git
2) Open TorchCraft and then BWEnv.sln (master)
3) Project>properties>VC++ Directories -- Add path to BWAPI/bwapi/include
4) also add czmq/include and libzmq/include
5) build

Errors with:
https://gist.github.com/Meriipu/70b80550267e83c812ed7d9aaa1955b6

I could not see the develop branch just now though

**Comment by ebetica** [comment]
Try:

right click on This PC -> Properties -> advanced system settings -> environment variables:

Set
BWAPI_DIR
to
C:\StarCraft\BWAPI

or whereever you installed BWAPI.

**Comment by Meriipu** [comment]
Seems to have solved it. Thanks.

**Comment by Meriipu** [comment]
No idea how agnostic the documentation should be to specific things like Visual Studio, but it would for me have been helpful if it mentioned the includes etc

**Comment by ebetica** [comment]
I'm not getting that issue, and I completely copy over TorchCraft every time I do a significant rebuild. Feel free to add a note to the right place and submit a PR :) The BWAPI_DIR stuff should have been set up if you installed BWAPI through their .exe.

**Comment by edran** [comment]
Reopening issue because the latest release of BWAPI does not setup `BWAPI_DIR` (just tried with a virgin VM).

**Comment by edran** [comment]
Fixed by #76.
